interpolate_phase.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python
+
+"""
+Copyright (c) 2023-, California Institute of Technology ("Caltech"). U.S.
+
+Government sponsorship acknowledged.
+All rights reserved.
+Author (s): Emre Havazli
+"""
+
+import argparse
+import glob
+import os
+import shutil
+import time
+
+import matplotlib.pyplot as plt
+import numpy as np
+from osgeo import gdal
+
+EXAMPLE = '''
+interpolate_phase.py <path_to_ifg_folder> <coherence_threshold>
+<interpolation_method>
+'''
+
+
+def create_parser():
+    """Create parser for command line arguments."""
+    parser = argparse.ArgumentParser(
+        description='Interpolate phase',
+        formatter_class=argparse.RawTextHelpFormatter,
+        epilog=EXAMPLE)
+
+    parser.add_argument(
+        'folder', type=str, help='Interferogram folder path')
+
+    parser.add_argument(
+        'coh_thresh', type=float, help='Coherence threshold to select '
+        'reliable pixels')
+
+    parser.add_argument(
+        'interp_method', type=str, help='Interpolation method: chen, chen_v2, numba')
+
+    return parser
+
+
+def get_intf(intf_file):
+    """Get interferogram and coherence from files."""
+    f = intf_file.split('/')
+    f[-1] = 'interp_filt_fine.int'
+    interp_file = os.path.join('/', *f)
+    shutil.copy(intf_file, interp_file)
+    shutil.copy(intf_file + '.xml', interp_file + '.xml')
+    ds = gdal.Open(interp_file, gdal.GA_ReadOnly)
+    print('Raster size=', ds.RasterXSize, ds.RasterYSize)
+    w_int = ds.GetRasterBand(1).ReadAsArray()
+    ds = None
+    print('interferogram dtype', w_int.dtype)
+
+    c = intf_file.split('/')
+    c[-1] = 'filt_fine.cor'
+    coh_file = os.path.join('/', *c)
+    ds = gdal.Open(coh_file, gdal.GA_ReadOnly)
+    coh = ds.GetRasterBand(1).ReadAsArray()
+    ds = None
+
+    # Plot Igram overview
+    print('Plotting interferogram overview...')
+    fig = plt.figure(figsize=(7, 5))
+    ax = fig.add_subplot(1, 2, 1)
+    cax = ax.imshow(np.angle(w_int), vmin=-3, vmax=3, cmap='hsv',
+                    interpolation='nearest')
+    ax.set_title(r'ifg phase')
+    cbar = fig.colorbar(cax, orientation='horizontal')
+    ax = fig.add_subplot(1, 2, 2)
+    cax = ax.imshow(coh, cmap='jet', interpolation='nearest')
+    ax.set_title(r'coh')
+    cbar = fig.colorbar(cax, orientation='horizontal')
+    plt.savefig('intf_overview.png', dpi=300, facecolor='white')
+    plt.close()
+
+    return interp_file, w_int, coh
+
+
+def get_ps_pixels(w_int, coh, coh_thresh):
```

Comment:
Maybe we should leave this as a more general function, right now we use a coherence threshold to generate our masks but maybe someone may want to use a different constraint or their own mask. Perhaps maybe we should check if a mask of reliable pixels was given and if not generate one based on a given (gdal readable layer) and a given threshold. What do you think?